solr/core/src/java/org/apache/solr/cloud/api/collections/CollectionHandlingUtils.java

```diff
@@ -135,6 +140,20 @@ public class CollectionHandlingUtils {
     }
   }
 
+  /** Returns names of properties that are used to specify a number of replicas of a given type. */
```

Comment:
But the reason for not doing it was to avoid exposing more code in solrj while it is only used in Solr itself.

Reply:
I see. But I'd still value more a cleaner object oriented design vs packaging/deployment concerns. SolrJ exposes a lot of internals to clients so within that framework I think it makes sense to group code related to the same concepts.

Reply:
I really would prefer not to bloat `Replica.Type` with methods, especially static ones, that are not used there. Since those methods are used only from API/Command layer, I believe that having them in a class named `CollectionHandlingUtils` makes sense. And since they are static (this is a different story for non-static methods, I added one as requested elsewhere in this PR) I do not see the difference it makes to have them here or there. So I definitely prefer not adding random util methods to a place where they do not belong.

Reply:
I don't understand why you say "do not belong".
Except standard Java classes, these two static methods depend only on `Replica.Type`.
If you don't want to put them in the `enum` put them in `Replica`, but move them close to the code they're related to.

Reply:
They do not belong there because they are not used in solrj, but in solr-core.